### PR TITLE
chore: trim a few imports and noshake exceptions

### PR DESCRIPTION
**Pull request by grunweg**
- remove a bogus import labelled as such (it is not necessary any more)
- remove a few now-superfluous exceptions from noshake.json.
- sort imports and remove a transitive import in Data/List/Basic

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
It builds, so it must be good

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

File: Mathlib/Data/Fintype/Basic.lean [hunk 1]
```diff
@@ -3,7 +3,6 @@ Copyright (c) 2017 Mario Carneiro. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Mario Carneiro
 -/
-import Mathlib.Algebra.Ring.Hom.Defs -- FIXME: This import is bogus
```
Comment (YaelDillies) [inline Mathlib/Data/Fintype/Basic.lean:6]: Note that `Algebra.Ring.Hom.Defs` is *still* imported in `Data.Fintype.Basic`. You're just hiding the problem.

**Comment by grunweg** [comment]
Let me close this: #12787 contains most of these changes; the bogus import removal is not helpful as it hides a problem.

**Pull request closed without merging**